packages/tendermint-rpc/src/dates.ts

```diff
@@ -25,6 +26,29 @@ export function toRfc3339WithNanoseconds(dateTime: ReadonlyDateWithNanoseconds):
   return `${millisecondIso.slice(0, -1)}${nanoseconds.padStart(6, "0")}Z`;
 }
 
+export function fromSeconds(seconds: number, nanos = 0): DateWithNanoseconds {
+  const checkedNanos = new Uint32(nanos).toNumber();
+  if (checkedNanos > 999_999_999) {
+    throw new Error("Nano seconds must not exceed 999999999");
+  }
+  const out: DateWithNanoseconds = new Date(seconds * 1000 + Math.floor(checkedNanos / 1000000));
+  out.nanoseconds = checkedNanos % 1000000;
+  return out;
+}
+
+/**
+ * Caclulates the UNIX timestamp in seconds as well as the nanoseconds after the given second.
+ *
+ * This is useful when dealing with external systems like the protobuf type
+ * [.google.protobuf.Timestamp](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp)
+ * or any other system that does not use millisecond precision.
+ */
+export function toSeconds(date: ReadonlyDateWithNanoseconds): [number, number] {
```

Comment:
Why not return a `Timestamp`?

Reply:
I wanted to have this function independent of protobuf generated code. There are too many implementation details, such as number/Long/string for the `seconds`.

We could return an object instead, but then it is a bit harder for the caller to rename the components. Now

```ts
function timestampFromDateNanos(date: ReadonlyDateWithNanoseconds): Timestamp {
  const [seconds, nanos] = toSeconds(date);
  return Timestamp.fromPartial({ seconds: Long.fromNumber(seconds), nanos });
}
```

with object

```ts
function timestampFromDateNanos(date: ReadonlyDateWithNanoseconds): Timestamp {
  const { seconds, nanoseconds: nanos } = toSeconds(date);
  return Timestamp.fromPartial({ seconds: Long.fromNumber(seconds), nanos });
}
```

But I'm open to this alternative.

Reply:
Then why not return `{seconds, nanos}` ? And standarize with a "Timestamp-like" format

Reply:
If the two of you prefer that over an unnamed tuples, this is fine for me. I don't have a real preference.

Reply:
I also don't have any real preference.

Reply:
I'll change to the object, which is easier to debug for our non-TS users